checker/src/visitors.rs

```diff
@@ -1775,6 +1776,14 @@ impl<'analysis, 'compilation, 'tcx, E> MirVisitor<'analysis, 'compilation, 'tcx,
                 };
                 qualifier.unwrap_or_else(|| abstract_value::BOTTOM.into())
             }
+            KnownFunctionNames::StdOpsDeref => {
+                checked_assume!(args.len() == 1);
+                args[0].1.deref().clone().into()
```

Comment:
Could you add a short comment why this is supposed to work? It appears you are mapping this to identity? Or is this to be refined later?

If its identity, does this work for every deref, also user defined?

Reply:
Good catch. deref is actually meant to be AbstractValue::dereference. The test case did not fail because there is some auto dereferencing going on elsewhere. I'm not sure I know how to quickly write a test case that will actually fail when mapping to identity, which is a bit sad.